bumpversion/__init__.py

```diff
@@ -336,7 +342,59 @@ def __iter__(self):
         return iter(self._values)
 
     def __repr__(self):
-        return '<bumpversion.Version:{}>'.format(keyvaluestring(self._values))
+        by_part = ", ".join("{}={}".format(k, v) for k, v in self.items())
+        return '<bumpversion.Version:{}>'.format(by_part)
+
+    def __hash__(self):
+        return hash(tuple((k, v) for k, v in self.items()))
+
+    def _compare(self, other, method, strict=True):
+        """
+        For strict relations, if a part is equal we skip it
+        :param other: the other Version
+        :param method: the compare method
+        :return:
+        """
+        try:
+            for vals in ((v, other[k]) for k, v in self.items()):
+                if vals[0] == vals[1]:
+                    continue
```

Comment:
IMHO it's cleaner to use avoid continue

Reply:
See previous comment on this method

Reply:
On a second pass I can not find a way to avoid the continue :/

Reply:
@arcanefoam Since the only purpose of the `continue` is to prevent calling `return`, you can just invert the logic.  Instead of

    for vals in ((v, other[k]) for k, v in self.items()):
        if vals[0] == vals[1]:
            continue
        return method(vals[0], vals[1])

Do
            
    for vals in ((v, other[k]) for k, v in self.items()):
        if vals[0] != vals[1]:
            return method(vals[0], vals[1])

I would also improve the readability by changing this to

    for key, value in self.items():
        own_part = vals[value]
        other_part = other[key]
        if own_part != other_part:
            return method(own_part, other_part)